deployment/cloudbuild-deploy.yaml

```diff
@@ -0,0 +1,140 @@
+---
+steps:
+  # Build the container image
+  - name: 'gcr.io/cloud-builders/docker'
+    entrypoint: '/bin/bash'
+    args:
+      - '-c'
+      - |
+        # Write namespace to file for future usage
+        case "$BRANCH_NAME" in
+          master)
+            echo "dev" > .namespace
+            ;;
+          staging)
+            echo "staging" > .namespace
+            ;;
+          stable)
+            echo "prod" > .namespace
+            ;;
+        esac
+
+        # Build container
+        docker build \
+          -f deployment/dockerfiles/$(cat .namespace)/${_CONTAINER} \
+          -t gcr.io/${PROJECT_ID}/$(cat .namespace)/${_CONTAINER}:${SHORT_SHA} \
+          --build-arg DEPLOY_TAG=${SHORT_SHA} \
+          .
+
+  # Push the origin-dapp container image
+  - name: 'gcr.io/cloud-builders/docker'
+    entrypoint: '/bin/bash'
+    args:
+      - '-c'
+      - |
+        docker push \
+          gcr.io/${PROJECT_ID}/$(cat .namespace)/${_CONTAINER}:${SHORT_SHA}"
+
+  # Add latest tag for to the container
+  - name: 'gcr.io/cloud-builders/gcloud'
+    entrypoint: 'sh'
```

Comment:
nit: should this be /bin/bash so that all entrypoints are consistent re what shell they use ?
same questions for other entrypoints below.

Reply:
Good point, I'll probably change them all to sh because I'm not sure all those containers have bash.